core/src/Kokkos_Parallel.hpp

```diff
@@ -132,8 +133,11 @@ template <
     class Enable = std::enable_if_t<is_execution_policy<ExecPolicy>::value>>
 inline void parallel_for(const std::string& str, const ExecPolicy& policy,
                          const FunctorType& functor) {
-  uint64_t kpID = 0;
+  /** Enforce correct use **/
+  Impl::CheckUsage<Impl::UsageRequires::insideExecEnv>::check(
+      Kokkos::ParallelForTag{}, str);
```

Comment:
It would be useful to pass information about the execution policy here as well since we have it.